src/components/Mission.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import styles from './Mission.module.css';
+import { joinMission, leaveMission } from '../redux/missions';
+
+const Mission = (props) => {
+  const dispatch = useDispatch();
+  const { mission } = props;
+  const {
+    id, name, description, reserved,
+  } = mission;
+  const handleMissionClick = () => {
+    if (mission.reserved === true) {
+      dispatch(leaveMission(id));
+    } else {
+      dispatch(joinMission(id));
+    }
+  };
```

Comment:
"Join Mission" button updates the store successfully by dispatching an action 💪😎 Well done ✔